### PR TITLE
ESB-126

### PR DESCRIPTION
**Pull request by solglass**
Swagger annotation for the lesson controller

### CODE REVIEW

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 1]
```diff
@@ -24,7 +24,11 @@ public LessonController(IMapper mapper, ILessonService lessonService)
             _lessonService = lessonService;
             _mapper = mapper;
         }
-
+        /// <summary>
+        /// Adds new lesson
+        /// </summary>
+        /// <param name="inputModel">Input model with all the properties for the new lesson</param>
+        /// <returns>Output model of the added lesson</returns>
         // https://localhost:50221/api/lesson/
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:32]: General comment: what about status codes?

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 2]
```diff
@@ -77,7 +98,11 @@ public ActionResult<LessonOutputModel> RecoverLesson(int id)
             return Ok(result);
         }
 
-
+        /// <summary>
+        /// Updates lesson's properties
+        /// </summary>
+        /// <param name="lessonId">Id of the lesson</param>
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:104]: only one param?

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 3]
```diff
@@ -89,7 +114,12 @@ public ActionResult<LessonOutputModel> UpdateLesson(int lessonId, [FromBody] Les
             var result = _mapper.Map<LessonOutputModel>(_lessonService.GetLessonById(lessonId));
             return Ok(result);
         }
-        
+
+        /// <summary>
+        /// Gets all feedbacks for the lesson
+        /// </summary>
+        /// <param name="lessonId">Id of the lesson</param>
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:121]: only one param?

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 4]
```diff
@@ -200,10 +279,10 @@ public ActionResult DeleteAttendance(int lessonId, int attendanceId)
 
         // https://localhost:50221/api/lesson/by-theme/14
         /// <summary>
-        /// Get all lessons that belong to one theme and are not deleted.
+        /// Gets all lessons that belong to one theme and are not deleted.
         /// </summary>
-        /// <param name="id">The identifier of the theme that we want to see all its lessons.</param>
-        /// <returns>The list of lessonOutputModel.</returns>
+        /// <param name="id">Id of the theme</param>
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:284]: rename to themeId

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 5]
```diff
@@ -161,6 +228,12 @@ public ActionResult<AttendanceOutputModel> GetAttendanceById(int lessonId, int a
...
         }
 
+        /// <summary>
+        /// Adds attendence for the lesson
+        /// </summary>
+        /// <param name="lessonId">Id of the lesson</param>
+        /// <param name="inputModel">Input model with all the properties for the attendence</param>
+        /// <returns>Added attendence output model </returns>
         // https://localhost:50221/api/lesson/5/attendance
         [HttpPost("{id}/attendance")]
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:238]: rename to lessonId

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 6]
```diff
@@ -151,6 +212,12 @@ public ActionResult<List<AttendanceOutputModel>> GetAttendancesByLessonId(int le
...
         }
 
+        /// <summary>
+        /// Gets attendence for the lesson by id
+        /// </summary>
+        /// <param name="lessonId">Id of the lesson</param>
+        /// <param name="attendanceId">Id of attendance</param>
+        /// <returns>Attendence output model for the lesson</returns>
         // https://localhost:50221/api/lesson/5/attendance/3
         [HttpGet("{id}/attendance/{attendanceId}")]
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:222]: rename to lessonId

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 7]
```diff
@@ -141,6 +197,11 @@ public ActionResult DeleteFeedback(int lessonId, int feedbackId)
             return StatusCode(StatusCodes.Status204NoContent);
         }
 
+        /// <summary>
+        /// Gets attendences for the lesson
+        /// </summary>
+        /// <param name="lessonId">Id of the lesson</param>
+        /// <returns>List of attendences output models for the lesson</returns>
         // https://localhost:50221/api/lesson/5/attendance/
         [HttpGet("{id}/attendance")]
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:206]: rename to lessonId

**Comment by solglass** [comment]
Fixed according to review notes

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 8]
```diff
@@ -35,61 +40,95 @@ public ActionResult<LessonOutputModel> AddNewLesson([FromBody] LessonInputModel
...
+        /// Gets all lessons of the Group
+        /// </summary>
+        /// <param name="GroupId">Group id </param>
+        /// <returns>List of Output models of the found Lessons </returns>
+        [ProducesResponseType(typeof(List<LessonOutputModel>), StatusCodes.Status200OK)]
+        // https://localhost:50221/api/lesson/45
+        [HttpGet("{GroupId}")]
         [Authorize(Roles = "Админ, Преподаватель, Студент")]
-        public ActionResult<List<LessonOutputModel>> GetLessons(int id)
+        public ActionResult<List<LessonOutputModel>> GetLessons(int GroupId)
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:52]: camelCase

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 9]
```diff
@@ -35,61 +40,95 @@ public ActionResult<LessonOutputModel> AddNewLesson([FromBody] LessonInputModel
...
 
+        /// <summary>
+        /// Gets lesson by id 
+        /// </summary>
+        /// <param name="lesonId"> Id of the lesson</param>
+        /// <returns>Output model of the found Lesson </returns>
+        [ProducesResponseType(typeof(LessonOutputModel), StatusCodes.Status200OK)]
         // https://localhost:50221/api/lesson/3
-        [HttpGet("{id}")]
+        [HttpGet("{lesonId}")]
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:66]: double "s"

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 10]
```diff
@@ -35,61 +40,95 @@ public ActionResult<LessonOutputModel> AddNewLesson([FromBody] LessonInputModel
...
+        /// Deletes lesson (soft-delete)
+        /// </summary>
+        /// <param name="lesonId"> Id of the lesson</param>
+        /// <returns>Output model of the soft-deleted Lesson</returns>
+        [ProducesResponseType(typeof(LessonOutputModel), StatusCodes.Status200OK)]
+        // https://localhost:50221/api/lesson/34
         [HttpDelete("{id}")]
         [Authorize(Roles = "Админ, Преподаватель")]
-        public ActionResult<LessonOutputModel> DeleteLesson(int id)
+        public ActionResult<LessonOutputModel> DeleteLesson(int lesonId)
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:83]: double "s"

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 11]
```diff
@@ -35,61 +40,95 @@ public ActionResult<LessonOutputModel> AddNewLesson([FromBody] LessonInputModel
...
             return Ok(result);
         }
 
-
-        // https://localhost:50221/api/lesson/id/recovery
-        [HttpPut("{id}/recovery")]
+        /// <summary>
+        /// Recovers deleted lesson
+        /// </summary>
+        /// <param name="lesonId"> Id of the lesson</param>
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:93]: double "s"

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 12]
```diff
@@ -141,6 +210,12 @@ public ActionResult DeleteFeedback(int lessonId, int feedbackId)
...
         }
 
+        /// <summary>
+        /// Gets attendences for the lesson
+        /// </summary>
+        /// <param name="lessonId">Id of the lesson</param>
+        /// <returns>List of attendences output models for the lesson</returns>
+        [ProducesResponseType(typeof(List<AttendanceOutputModel>), StatusCodes.Status200OK)]
         // https://localhost:50221/api/lesson/5/attendance/
         [HttpGet("{id}/attendance")]
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:220]: still has to be renamed

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 13]
```diff
@@ -151,6 +226,13 @@ public ActionResult<List<AttendanceOutputModel>> GetAttendancesByLessonId(int le
...
 
+        /// <summary>
+        /// Gets attendence for the lesson by id
+        /// </summary>
+        /// <param name="lessonId">Id of the lesson</param>
+        /// <param name="attendanceId">Id of attendance</param>
+        /// <returns>Attendence output model for the lesson</returns>
+        [ProducesResponseType(typeof(AttendanceOutputModel), StatusCodes.Status200OK)]
         // https://localhost:50221/api/lesson/5/attendance/3
         [HttpGet("{id}/attendance/{attendanceId}")]
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:237]: still has to be renamed

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 14]
```diff
@@ -161,6 +243,13 @@ public ActionResult<AttendanceOutputModel> GetAttendanceById(int lessonId, int a
...
 
+        /// <summary>
+        /// Adds attendence for the lesson
+        /// </summary>
+        /// <param name="lessonId">Id of the lesson</param>
+        /// <param name="inputModel">Input model with all the properties for the attendence</param>
+        /// <returns>Added attendence output model </returns>
+        [ProducesResponseType(typeof(AttendanceOutputModel), StatusCodes.Status200OK)]
         // https://localhost:50221/api/lesson/5/attendance
         [HttpPost("{id}/attendance")]
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:254]: still has to be renamed

**Comment by solglass** [comment]
Fixed again

**Pull request merged**